### ISSUE TITLE
Compiler generates bad code if base class has an inner class named Type

### ISSUE THREAD

**Issue by ZacSweers** [rharter/auto-value-moshi#26]
Sort of an edge case, but putting it on your radar.

``` java
public abstract class A {

  public enum Type { Apple, Banana }

  abstract Type type();
}
```

Resulting error mentions something like:

`Error:(28, 23) error: AutoValue_AJsonAdapterFactory is not abstract and does not override abstract method create(Type,Set<? extends Annotation>,Moshi) in Factory`

Because the `Type` in that method is incorrectly pointing to the inner class rather than `java.lang.reflect.Type`

**Comment by rharter** [comment]
This is an issue with JavaPoet and is tracked in this issue. https://github.com/square/javapoet/issues/77

I'll leave this open until we come up with a valid solution.

**Comment by ZacSweers** [comment]
This is resolved in javapoet 1.12.0 now
